addon/v1alpha1/types_addondeploymentconfig.go

```diff
@@ -0,0 +1,74 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AddOnDeploymentConfig represents a deployment configuration for an add-on.
+// AddOnDeploymentConfig is a cluster-scoped resource.
+type AddOnDeploymentConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec represents a desired configuration for an add-on.
+	// +required
+	Spec AddOnDeploymentConfigSpec `json:"spec"`
+}
+
+type AddOnDeploymentConfigSpec struct {
+	// CustomizedVariables is a list of name-value variables for the current add-on deployment.
+	// The add-on implementation can use these variables to render its add-on deployment.
+	// The default is an empty list.
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	CustomizedVariables []CustomizedVariable `json:"customizedVariables,omitempty"`
+
+	// NodePlacement enables explicit control over the scheduling of the add-on agents on the
+	// managed cluster.
+	// All add-on agent pods are expected to comply with this node placement.
+	// +optional
+	NodePlacement NodePlacement `json:"nodePlacement,omitempty"`
+}
+
+// CustomizedVariable represents a customized variable for add-on deployment.
+type CustomizedVariable struct {
+	// Name of this variable.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// Value of this variable.
+	Value string `json:"value"`
```

Comment:
you may want to add a maxlength validation.

Reply:
limit the length to 255